### ISSUE TITLE
Env detection not working as expected

### ISSUE THREAD

**Issue by baztian** [ryanhiebert/tox-travis#62]
I'm trying to a travis matrix to tox.
`.travis.yml`

```yaml
python:
  - '2.6'
  - '2.7'
  - '3.4'

env:
  matrix:
    include:
      - python: 3.4
        env: JYTHON="org.python;jython-installer;2.7.0"
```

`tox.ini`

```ini
[tox]
envlist = py{26,27,3}-driver-{hsqldb,mock,sqliteXerial},py27-driver-sqlitePy,jython-driver-{hsqldb,mock}

[travis]
python =
  2.6: py26-driver-{hsqldb,mock,sqliteXerial}
  2.7: py27-driver-{hsqldb,mock,sqliteXerial,sqlitePy}
  3.4: py34-driver-{hsqldb,mock,sqliteXerial}

[travis:env]
JYTHON =
  org.python;jython-installer;2.7.0: jython-driver-{hsqldb,mock}
```

(You can find the real code in [my feature branch](/baztian/jaydebeapi/tree/travis-tox))

[The build](https://travis-ci.org/baztian/jaydebeapi/builds/209340861) runs fine (the error is not related to tox mapping problems). But the Python 3.4 + Jython-Env entry doesn't get mapped as desired. It maps to all tox environments whereas I was expecting it to match to just the Jython environments (`jython-driver-{hsqldb,mock}`).

Am I doing something wrong or is this a bug?

**Comment by ryanhiebert** [comment]
It took me a bit to figure out what it was. I finally spotted the problem when I was looking at the diff of the branch on your repository. I spotted it because of GitHub's syntax highlighting. So I also added it above. The main issue is that the `;` character is one of the comment characters for `ini` (along with `#`), and your configuration isn't even getting fully read.

However, even with that fixed you're going to have some problems with your configuration. It won't run anything on the jython variant, because the `python = 3.4` variant configured there will _also_ be applied, not merely the variant for the `JYTHON` environment variable. All of the configuration must match, so it won't work the way you're wanting.

I think that the way I'd implement your configuration is to drop the `travis:env` section, and add `TOXENV="jython-driver-{hsqldb,mock}"` to that environment configuration in `.travis.yml`. I think that would do what you're looking for.
